src/magic_folder/uploader.py

```diff
@@ -434,11 +435,28 @@ def startService(self):
 
         # do a looping call that polls the db for LocalSnapshots.
         self._processing_loop = LoopingCall(
-            self._remote_snapshot_creator.upload_local_snapshots,
+            self.perform_upload
         )
         self._processing_loop.clock = self._clock
         self._processing = self._processing_loop.start(self._poll_interval, now=True)
 
+
+    def perform_upload(self):
+        """
+        Do an upload unless we are already doing one.
+        """
+        if self._uploading:
```

Comment:
I do not think that makes sense. That would allow pending upload tasks to stack up indefinitely.

Reply:
(At the very least, the "timer triggered" update _shouldn't_ get stacked up, it should just skip .. )

Reply:
Well, `LoopingCall` doesn't stack up calls, so we'll only ever get a single extra stacked call from that, if the timer triggers while a manually triggered upload is running.

But we do want at least one (and preferably exactly one) pending upload task to stack up, if they are explicitly requested, since the list of snapshots needing to uploaded will be out-of-date in the running one, if we get a new request.

Reply:
They'll still stack up, but only if the uploads take longer than `scan-interval`

Reply:
> They'll still stack up, but only if the uploads take longer than `scan-interval`

The only time a periodic call will stack up, is if came in while a explicitly requested call was running. While a periodic call is waiting for the lock, there isn't delayed call that will trigger a new periodic call; that delayed call is only created once the first periodic call has acquired the lock and completed its work.

In any case, I filed #552 to implement a new primitive that will ensure calls don't ever stack up.